mlflow/tracking/fluent.py

```diff
@@ -1348,8 +1350,13 @@ def _get_experiment_id():
     return (
         _active_experiment_id
         or _get_experiment_id_from_env()
-        or (is_in_databricks_notebook() and get_notebook_id())
         or (is_in_databricks_job() and get_job_type_info() == "NORMAL" and _create_job_experiment())
+        or (
+            is_in_databricks_notebook()
+            and is_in_databricks_repo_notebook()
+            and _create_repo_notebook_experiment()
+        )
+        or (is_in_databricks_notebook() and get_notebook_id())
```

Comment:
Can we get https://github.com/mlflow/mlflow/pull/5309 merged first, which would simplify this logic? cc @sunishsheth2009

Reply:
I was going to work on that next sprint but I can work on it this sprint instead. Let me work on it now.